pkg/controller/gkenetworkparamset/gkenetworkparamset_controller_test.go

```diff
@@ -1154,3 +1180,441 @@ func (testVals *testGKENetworkParamSetController) doesGNPFinalizerExist(ctx cont
 
 	return false, nil
 }
+
+func TestPopulateDesiredDefaultParamSet(t *testing.T) {
+	desiredDefaultParamSet := newL3GNP(networkv1.DefaultPodNetworkName, ensureExistsMode, []string{defaultPodRange})
+	invalidDefaultParamSet := newL3GNP(networkv1.DefaultPodNetworkName, reconcileMode, nil)
+
+	tests := []struct {
+		name            string
+		defaultParamSet *networkv1.GKENetworkParamSet
+		wantParamSet    *networkv1.GKENetworkParamSet
+	}{
+		{
+			name: "not populate if addon is reconcile mode",
+			defaultParamSet: &networkv1.GKENetworkParamSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: networkv1.DefaultPodNetworkName,
+					Annotations: map[string]string{
+						annotationComponentsLayer: componentLayer,
+						annotationComponentsName:  componentName,
+					},
+					Labels: map[string]string{
+						labelsAddonManagerMode: reconcileMode,
+					},
+				},
+				Spec: networkv1.GKENetworkParamSetSpec{
+					VPC:       defaultTestNetworkName,
+					VPCSubnet: defaultTestSubnetworkName,
+				},
+			},
```

Comment:
It is much more readable to create a builder function that takes as input the VPC, subnet and pod ranges and returns this GKENEtworkParamSet object.

Reply:
You can call newL3GNP here right?

Reply:
Done. I modify the newL3GNP a bit.